test/JLM_Tests/src/org/openj9/test/java/lang/management/TestOpenJ9DiagnosticsMXBean.java

```diff
@@ -0,0 +1,569 @@
+
+		for (String agent : dumpAgents) {
+			try {
+				String filePath = dir + File.separator + files[index];
+
+				String fileName = diagBean.triggerDumpToFile(agent, filePath);
+				File dumpFile = new File(fileName);
+
+				if (!(agent.equals("stack"))) {
+					found = findFile(dir, dumpFile.getName());
+					AssertJUnit.assertTrue(fileName + " not found", found);
+					
+				} 
+			
+			} catch (java.lang.IllegalArgumentException e) {
+				/* stack agent is not supported by the trigger method */
+				if (agent.equals("stack")) {
+					AssertJUnit.assertTrue("Did not receive the expected exception IllegalArgumentException", true);
+				} 
+				else {
+					Assert.fail("Unexpected IllegalArgumentException : " + e.getMessage());
+				}
+			} catch (Exception e) {
+				Assert.fail("Unexpected Exception thrown : " + e.getMessage());
+				e.printStackTrace();
+			}
+
+			index++;
+		}
+	}
+
+	/**
+	 * Function to test if a class heap dump is created
+	 *
+	 * @param diagBean - OpenJ9DiagnosticsMXBean instance that has already been initialized
+	 */	
+	private static void test_triggerClassicHeapDump(OpenJ9DiagnosticsMXBean diagBean) {
+		boolean found = false;
+
+		try {
+			String fileName = diagBean.triggerClassicHeapDump();
+			found = findFile(".", new File(fileName).getName());
+			AssertJUnit.assertTrue(fileName + " not found", found);
+		} catch (InvalidDumpOptionException e){
+			Assert.fail("Unexpected InvalidDumpOptionException : " + e.getMessage());
+		} catch (Exception e){
+			Assert.fail("Unexpected Exception thrown : " + e.getMessage());
+			e.printStackTrace();
+		}
+
+	}	
+
+
+	/**
+	 * Function to test if the dump options were reset to what it was at JVM initialization
+	 *
+	 * @param diagBean - OpenJ9DiagnosticsMXBean instance that has already been initialized
+	 */	
+	private void test_resetDumpOption(OpenJ9DiagnosticsMXBean diagBean) {
+		boolean found = false;
+		try {
+			diagBean.setDumpOptions("java+heap+system:events=vmstop");			
+		} catch (InvalidDumpOptionException e) {
+			Assert.fail("Unexpected InvalidDumpOptionException : " + e.getMessage());
+		} catch (Exception e) {
+			Assert.fail("Unexpected Exception thrown : " + e.getMessage());
+		}		
+		
+	
+		try {	
+			diagBean.resetDumpOption();
+		} catch (DumpConfigurationUnavailableException e) {
+			Assert.fail("Unexpected DumpConfigurationUnavailableException : " + e.getMessage());
+		}
+
+		List<String> newDumpOptions = new ArrayList<String>();
+		newDumpOptions = getDumpOptions();
+		boolean res = initialDumpOptions.equals(newDumpOptions);
+		
+		AssertJUnit.assertTrue("Reset Dump Options failed", res);
+		
+	}
+
+	
+	/**
+	 * Function to check if the specified file is present in the directory path 
+	 *
+	 * @param dirPath - Directory path where dumps are created 
+	 * @param fileName - Dump file name
+	 * @return a boolean indicating if the file is found or not
+	 */	
+	private static boolean findFile(String dirPath, String fileName) {
```

Comment:
Should this method be called findAndDeleteFile?  since it appears that is what the method does